packages/drivers/odsp-driver-web-cache/src/test/FluidCache.test.ts

```diff
@@ -0,0 +1,288 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ICacheEntry } from "@fluidframework/odsp-driver-definitions";
+import { openDB } from "idb";
+import { FluidCache } from "../FluidCache";
+import {
+    getFluidCacheIndexedDbInstance,
+    FluidDriverObjectStoreName,
+    FluidDriverCacheDBName,
+    getKeyForCacheEntry,
+} from "../FluidCacheIndexedDb";
+// eslint-disable-next-line max-len
+// eslint-disable-next-line import/no-unassigned-import, @typescript-eslint/no-require-imports, import/no-internal-modules
+require("fake-indexeddb/auto");
+
+const mockPartitionKey = "FAKEPARTITIONKEY";
+
+function getFluidCache(config?: {
+    maxCacheItemAge?: number;
+    // eslint-disable-next-line @rushstack/no-new-null
+    partitionKey?: string | null;
+}) {
+    return new FluidCache({
+        partitionKey: config?.partitionKey || mockPartitionKey,
+        maxCacheItemAge: config?.maxCacheItemAge || 3 * 24 * 60 * 60 * 1000,
+    });
+}
+
+class DateMock {
+    // The current time being used by the mock
+    public static mockTimeMs: number = 0;
+
+    public static now() {
+        return DateMock.mockTimeMs;
+    }
+
+    public getTime() {
+        return DateMock.mockTimeMs;
+    }
+}
+
+// Sets up a mock date time for the current test. Returns a function that should be called to reset the environment
+function setupDateMock(startMockTime: number) {
+    const realDate = window.Date;
```

Comment:
I assume there is no way to run this test (and shipping code in general) in Node environment, as there is no equivalence for indexDb, right? Most of our code is isomorphic, and it would be great if we can use and test this functionality in our stress tests (they are run in node environment)

Reply:
Yeah this has to run in a browser or mock browser environment. A stress test for this would need to run in a browser context.